### PR TITLE
Add SyncPod() to DockerManager and use it in Kubelet

### PR DESCRIPTION
**Pull request by vmarmol**
/cc @yujuhong @yifan-gu @dchen1107 

@brendandburns this is the last big refactoring that will affect you.

Unit and integration tests pass, running e2e now.

### CODE REVIEW

**Comment by yujuhong** [comment]
Looks good overall. Will make another pass in the morning.

**Comment by vmarmol** [comment]
A couple of e2e tests are failing. Will investigate more in the morning.

File: pkg/kubelet/dockertools/manager.go [hunk 1]
```diff
@@ -1315,3 +1334,97 @@ func (dm *DockerManager) ComputePodContainerChanges(pod *api.Pod, runningPod kub
...
+		return nil
+	}
+
+	err = dm.PullImage(container.Image)
+	dm.runtimeHooks.ReportImagePull(pod, container, err)
+	return err
+}
+
+// Sync the running pod to match the specified desired pod.
+func (dm *DockerManager) SyncPod(pod *api.Pod, runningPod kubecontainer.Pod, podStatus api.PodStatus) error {
```
Comment (yujuhong) [inline pkg/kubelet/dockertools/manager.go]: Just wanted to make sure: the code in SyncPod was moved from kubelet, so I am assuming it's safe. Is there any change that's worth paying more attention to?

**Comment by yujuhong** [comment]
LGTM.

**Comment by vmarmol** [comment]
@yujuhong yes the code is almost completely a direct move and hanging from calling `Kubelet` methods to `DockerManager` equivalents.

The only major change I can think of is that we used to:
- calculate changes
- mount volumes
- apply changes

and now we:

- mount volumes
- calculate changes
- apply changes

I am seeing the DNS e2e failing, but I'm unsure if it is this or just a flake.

**Comment by yujuhong** [comment]
I doubt that the change would cause DNS failing, but it'd be nice to use ginkgo to rerun only the DNS test to make sure (given that e2e has been green for a while).

**Comment by vmarmol** [comment]
That is my thought as well, a couple of other e2e's are failing as well, but those look like it is due to DNS so they may all be connected.

**Comment by yifan-gu** [comment]
e2e is failing those tests since the wed. FWIW https://github.com/GoogleCloudPlatform/kubernetes/issues/7548

**Comment by yujuhong** [comment]
@yifan-gu, e2e has been green since last night though :(

**Comment by yujuhong** [comment]
I ran two rounds of e2e with this PR, and both passed without any problem. I think it's safe to merge this PR. Will wait until either shippable or travis is green just in case.

**Comment by vmarmol** [comment]
Thanks for verifying @yujuhong!

**Comment by yujuhong** [comment]
Shippable is green. Merging :)

**Comment by vmarmol** [comment]
:D

**Pull request merged**